SYCL/OptionalKernelFeatures/is_compatible.cpp

```diff
@@ -61,7 +61,7 @@ int main() {
   }
 
   if (Dev.get_info<sycl::info::device::max_work_group_size>() > INT_MAX) {
-    Compatible &= true;
+    Compatible &= false;
```

Comment:
I'm a bit confused about the intent behind making this check separately from `is_compatible`. If we do expect such a case, why do we have an unconditional assertion failure in the `is_compatible` branch below? Shouldn't we assert that `Dev.get_info<sycl::info::device::max_work_group_size>() >= INT_MAX == is_compatible...` instead?